py-polars/polars/internals/frame.py

```diff
@@ -1519,6 +1519,36 @@ def to_numpy(self) -> np.ndarray:
         else:
             return out
 
+    def _comp(
+        self: DF, other: Any, op: Literal["eq", "neq", "gt", "lt", "gt_eq", "lt_eq"]
+    ) -> DF:
+        if isinstance(other, DataFrame):
+            if self.columns != other.columns:
+                raise ValueError("DataFrame columns do not match")
+            data_series = [self[c]._comp(other[c], op) for c in self.columns]
```

Comment:
We can parallelize this by doing a horizontal `concat` with a `suffix()` and then run 

`pl.all() == pl.all().suffix(<suffix_name>)`

Reply:
I figured there would be a better way, thanks for the input! I can't exactly figure out your pointer though.

### Comparing with a non-DataFrame
For comparing a DataFrame to a **non**-DataFrame, we can do something like:

```python
return self.select(pl.all() > other)
```
This would indirectly dispatch to the `Series._comp` method, and use the polars query engine, which should be faster than my list comprehension. But how do I then nicely do this for all comparison types? I tried a `map` with `pl.Series._comp` but that doesn't seem to work.

### Comparing with a DataFrame

I get the idea of doing a horizontal concat and then somehow doing an expression that calls the Series comparisons in parallel. But first off, it seems that I cannot do a horizontal concat lazily:

```python
self = pl.DataFrame({"a": [1, 2], "b": [3, 4]})
other = pl.DataFrame({"a": [1, 2], "b": [3, 5]})
df_big = pl.concat([self.lazy(), other.lazy().select(pl.all().suffix('__other'))], how='horizontal')
df_big.collect()
# SchemaError: cannot vstack: because column names in the two DataFrames do not match for left.name='a' != right.name='a__other'
```

It tries to concatenate vertically, even though we specify 'horizontal'. If you check the code for `concat`, the `how` argument is disregarded for LazyFrames. Is this intentional? And if so, are you proposing we perform the horizontal concat eagerly? Because it seems very memory inefficient to build such a big DataFrame.

Secondly, if we do manage to build the concatenated DataFrame, I tried doing something like:
```python
df_big.select(pl.col(['a', 'b']) == pl.col(['a__other', 'b__other']))
# shape: (2, 2)
# ┌──────────┬──────────┐
# │ a__other ┆ b__other │
# │ ---      ┆ ---      │
# │ bool     ┆ bool     │
# ╞══════════╪══════════╡
# │ true     ┆ true     │
# ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌┤
# │ true     ┆ true     │
# └──────────┴──────────┘
```
But the result is incorrect, so it's not nicely comparing each column. Could you elaborate your suggestion, maybe?

Reply:
All right, I implemented it like this!